developer_docs/foreman_dev_setup.asciidoc

```diff
@@ -0,0 +1,138 @@
+Before installing PostgreSQL, check first that you don't already have it installed (by running `rpm -q postgresql`).
+
+You will want to make sure that you have postgresql-devel installed so the database gems can install properly.
+Also, you would also need gcc, ruby-devel, libxml-devel, libxslt-devel and libvirt-devel packages.
+
+For Fedora, CentOS or RHEL, you can issue the following commands to install all required packages:
+[source, bash]
+....
+dnf groupinstall "Development Tools" "Development Libraries"
+dnf -y install gcc-c++ git ruby ruby-devel rubygems \
+    libvirt-devel postgresql-devel openssl-devel \
+    libxml2-devel libxslt-devel zlib-devel \
+    readline-devel systemd-devel tar libcurl-devel
+....
+
+[[Setup]]
+== Setting the repository
+[source, bash]
+....
+# On Github create your own fork
+git clone https://github.com/{GITHUB_USER}/foreman.git -b develop
+cd foreman
+cp config/settings.yaml.example config/settings.yaml
+cp config/database.yml.example config/database.yml
+git remote add upstream git@github.com:theforeman/foreman.git
+gem install bundler
+bundle install
+npm install
+....
+
+[[Database]]
+== Setting the database
+It is important that config/database.yml is set to use the correct database in the “development” block.
+Rails (and subsequently Foreman) will use these connection settings under “development” to manage the database it uses and setup the necessary schema.
+
+If the default version of the database.yml file isn't working for you, try editing it with the following:
+[source, ruby]
+....
+development:
+ adapter: postgresql
+ host: localhost
+ username: postgres
+ password: password
+ pool: 5
+ database: foreman_development
+
+test:
+ adapter: postgresql
+ host: localhost
+ username: postgres
+ password: password
+ pool: 5
+ database: foreman_test
+....
+
+Set up database schema, precompile assets and locales:
+[source, ruby]
+....
+bundle exec rake db:create # if not already created
+bundle exec rake db:migrate
+bundle exec rake db:seed
+....
+
+The db:seed step will print out the default admin password, record this in order to log in later.
+
+The previous commands will create databases only for the development environment, for test environment you need to run also:
+[source, ruby]
+....
+RAILS_ENV=test bundle exec rake db:create
+RAILS_ENV=test bundle exec rake db:migrate
+....
+
+[[Running]]
+== Running Foreman
+In order to run Foreman you can use the following command inside your git repository:
+[source, bash]
+....
+bundle exec foreman start
+....
+
+You can also run Foreman in two separate processes - frontend and rails. This way when you need to restart Foreman (for code reload) you don't have to wait for webpack build:
+[source, bash]
+....
+# Rails
+bundle exec rails s -b 0.0.0.0 -p 3000
+
+# Frontend
+env NODE_ENV=development NOTIFICATIONS_POLLING=99999999 ./node_modules/.bin/webpack-dev-server-without-h2 --config ./config/webpack.config.js
```

Comment:
I think we should only include the necessary parameters here, so only keep:  `./node_modules/.bin/webpack-dev-server-without-h2 --config config/webpack.config.js`
And explain `NODE_ENV` in the lines under, like with `NOTIFICATIONS_POLLING`